src/SmartComponents/modules/reducers.js

```diff
@@ -1,14 +1,65 @@
 import types from './types';
 
 const initialState = {
-    compare: {},
+    fullCompareData: {},
     addSystemModalOpened: false,
     selectedSystemIds: [],
     filterDropdownOpened: false,
     stateFilter: 'all',
-    factFilter: ''
+    factFilter: '',
+    totalFacts: 0,
+    page: 1,
+    perPage: 10,
+    filteredCompareData: {}
 };
 
+function paginateFilteredData(data, selectedPage, factsPerPage) {
+    let paginatedData = {};
+    let paginatedFacts = [];
+    let divisor = 0;
+
+    for (let i = 0; i < data.facts.length; i++) {
+        divisor++;
```

Comment:
This means that divisor starts at `1` and is incremented on every iteration right? The `i` variable starts at `0` and increments every iteration. Feels like you could simplify it down to one variable?

Reply:
yes, but I know it's not standard to modify the for loop counter and it is standard to start at 0. But, if you're good with it, I'll just do math on i+1.

Reply:
Sounds good to me